packages/dev/core/src/Materials/Node/nodeMaterialBuildState.ts

```diff
@@ -538,4 +538,74 @@ export class NodeMaterialBuildState {
             return `${this._getShaderType(type)} ${name}`;
         }
     }
+
+    private _convertVariableDeclarationToWGSL(type: string, dest: string, source: string): string {
+        return source.replace(new RegExp(`(${type})\\s+(\\w+)`, "g"), `var $2: ${dest}`);
+    }
+
+    private _convertVariableConstructorsToWGSL(type: string, dest: string, source: string): string {
+        return source.replace(new RegExp(`(${type})\\(`, "g"), ` ${dest}(`);
+    }
+
+    private _convertOutParametersToWGSL(source: string): string {
+        return source.replace(new RegExp(`out\\s+var\\s+(\\w+):\\s+(\\w+)`, "g"), `$1: ptr<function, $2>`);

```

Comment:
Maybe you should allow optional whitespaces after the identifier name and before the semi-colon, and also allow no whitespaces after the semi-colon?

Like:
```typescript
new RegExp(`out\\s+var\\s+(\\w+)\\s*:\\s*(\\w+)`, "g")`
```

Reply:
I agree that will make it more flexible